### PR TITLE
gh-92932: dis._unpack_opargs should handle EXTENDED_ARG_QUICK

### LINKED ISSUE #92932: dis._unpack_opargs should handle EXTENDED_ARG_QUICK
In Python 3.11b0, `dis.dis()` and `dis.get_instructions()` will show the "wrong" operand values for instructions prefixed by EXTENDED_ARG_QUICK. 

https://github.com/python/cpython/blob/a4460f2eb8b9db46a9bce3c450c8b038038a7c93/Lib/dis.py#L591-L595

The bug is that this line only checks for `EXTENDED_ARG` instructions. It should check for instructions that deoptimize to `EXTENDED_ARG` (i.e. `deop == EXTENDED_ARG`).

https://github.com/python/cpython/blob/a4460f2eb8b9db46a9bce3c450c8b038038a7c93/Lib/dis.py#L595

To reproduce the issue, consider this snippet adapted from cloudpickle:

```python
import random
import textwrap
import dis

nvars = 65537 + 258
names = ['g%d' % i for i in range(1, nvars)]
r = random.Random(42)
d = {name: r.randrange(100) for name in names}
# def f(x):
#     x = g1, g2, ...
code = """
def f():
    x = {tup}
""".format(tup=', '.join(names))
exec(textwrap.dedent(code), d, d)
f = d['f']

dis.dis(f)
```

### PR DESCRIPTION
**Pull request by corona10**
closes: gh-92932

### CODE REVIEW

**Comment by Fidget-Spinner** [comment]
Sorry Dong-hee, I'm useless at `dis`. Can't help you review this here :).

**Comment by corona10** [comment]
`````diff

230261,230262c230261,230262
<            1118398 EXTENDED_ARG_QUICK       1
<            1118400 LOAD_GLOBAL            248 (g125)
---
>            1118398 EXTENDED_ARG_QUICK     513
>            1118400 LOAD_GLOBAL          131576 (g65789)
230265,230266c230265,230266
<            1118416 EXTENDED_ARG_QUICK       1
<            1118418 LOAD_GLOBAL            250 (g126)
---
>            1118416 EXTENDED_ARG_QUICK     513
>            1118418 LOAD_GLOBAL          131578 (g65790)
230269,230270c230269,230270
<            1118434 EXTENDED_ARG_QUICK       1
<            1118436 LOAD_GLOBAL            252 (g127)
---
>            1118434 EXTENDED_ARG_QUICK     513
>            1118436 LOAD_GLOBAL          131580 (g65791)
230273,230274c230273,230274
<            1118452 EXTENDED_ARG_QUICK       1
<            1118454 LOAD_GLOBAL            254 (g128)
---
>            1118452 EXTENDED_ARG_QUICK     513
>            1118454 LOAD_GLOBAL          131582 (g65792)
230277,230278c230277,230278
<            1118470 EXTENDED_ARG_QUICK       2
<            1118472 LOAD_GLOBAL              0 (g1)
---
>            1118470 EXTENDED_ARG_QUICK     514
>            1118472 LOAD_GLOBAL          131584 (g65793)
230281,230282c230281,230282
<            1118488 EXTENDED_ARG_QUICK       2
<            1118490 LOAD_GLOBAL              2 (g2)
---
>            1118488 EXTENDED_ARG_QUICK     514
>            1118490 LOAD_GLOBAL          131586 (g65794)
`````

File: Lib/dis.py [hunk 1]
```diff
@@ -592,7 +592,7 @@ def _unpack_opargs(code):
         caches = _inline_cache_entries[deop]
         if deop >= HAVE_ARGUMENT:
             arg = code[i+1] | extended_arg
-            extended_arg = (arg << 8) if op == EXTENDED_ARG else 0
+            extended_arg = (arg << 8) if EXTENDED_ARG in (op, deop) else 0
```
Comment (brandtbucher) [inline Lib/dis.py]:
```suggestion
            extended_arg = (arg << 8) if deop == EXTENDED_ARG else 0
```

**Review by brandtbucher — CHANGES_REQUESTED** [review]
Thanks for the fix!

This should also have a test in `test_dis`. As a simple example, the statement `*_, _ = ...` should always generate an instruction `UNPACK_EX(256)` (proving that `EXTENDED_ARG` works correctly).

**Comment by corona10** [comment]
I have made the requested changes; please review again

**Comment by corona10** [comment]
@brandtbucher gentle ping :)

File: Lib/test/test_dis.py [hunk 2]
```diff
@@ -997,6 +1013,11 @@ def test_loop_quicken(self):
         got = self.get_disassembly(loop_test, adaptive=True)
         self.do_disassembly_compare(got, dis_loop_test_quickened_code)
 
+    @cpython_only
+    def test_extenaded_arg_quick(self):
```
Comment (brandtbucher) [inline Lib/test/test_dis.py]:
```suggestion
    def test_extended_arg_quick(self):
```

File: Lib/test/test_dis.py [hunk 3]
```diff
@@ -620,6 +620,22 @@ def loop_test():
...
+
+%3d           2 LOAD_CONST               1 (Ellipsis)
+              4 EXTENDED_ARG_QUICK       1
+              6 UNPACK_EX              256
+              8 STORE_FAST               0 (_)
+             10 STORE_FAST               0 (_)
+             12 LOAD_CONST               0 (None)
+             14 RETURN_VALUE
+"""% (extend_arg_quick.__code__.co_firstlineno,
+      extend_arg_quick.__code__.co_firstlineno + 1,)
```
Comment (brandtbucher) [inline Lib/test/test_dis.py]:
```suggestion
def extended_arg_quick():
    *_, _ = ...

dis_extended_arg_quick_code = """\
%3d           0 RESUME                   0

%3d           2 LOAD_CONST               1 (Ellipsis)
              4 EXTENDED_ARG_QUICK       1
              6 UNPACK_EX              256
              8 STORE_FAST               0 (_)
             10 STORE_FAST               0 (_)
             12 LOAD_CONST               0 (None)
             14 RETURN_VALUE
"""% (extended_arg_quick.__code__.co_firstlineno,
      extended_arg_quick.__code__.co_firstlineno + 1,)
```

File: Lib/test/test_dis.py [hunk 4]
```diff
@@ -997,6 +1013,11 @@ def test_loop_quicken(self):
         got = self.get_disassembly(loop_test, adaptive=True)
         self.do_disassembly_compare(got, dis_loop_test_quickened_code)
 
+    @cpython_only
+    def test_extenaded_arg_quick(self):
+        got = self.get_disassembly(extend_arg_quick)
```
Comment (brandtbucher) [inline Lib/test/test_dis.py]:
```suggestion
        got = self.get_disassembly(extended_arg_quick)
```

**Review by brandtbucher — APPROVED** [review]
Looks good! I found a couple of spelling nits, but feel free to merge otherwise:

**Comment by corona10** [comment]
Thank you Brandt for review!

**Pull request merged**